common/libzkp/impl/src/tasks/batch.rs

```diff
@@ -0,0 +1,253 @@
+use crate::proofs::ChunkProof;
+use c_kzg::Bytes48;
+use eyre::Result;
+use sbv_primitives::{B256, U256};
+use scroll_zkvm_types::{
+    batch::{
+        BatchHeader, BatchHeaderV6, BatchHeaderV7, BatchInfo, BatchWitness, EnvelopeV6, EnvelopeV7,
+        PointEvalWitness, ReferenceHeader, N_BLOB_BYTES,
+    },
+    public_inputs::ForkName,
+    task::ProvingTask,
+    utils::{to_rkyv_bytes, RancorError},
+};
+
+mod utils;
+use utils::{base64, point_eval};
+
+/// Define variable batch header type, since BatchHeaderV6 can not
+/// be decoded as V7 we can always has correct deserialization
+/// Notice: V6 header MUST be put above V7 since untagged enum
+/// try to decode each defination in order
+#[derive(Clone, serde::Deserialize, serde::Serialize)]
+#[serde(untagged)]
+pub enum BatchHeaderV {
+    V6(BatchHeaderV6),
+    V7(BatchHeaderV7),
+}
+
+impl From<BatchHeaderV> for ReferenceHeader {
+    fn from(value: BatchHeaderV) -> Self {
+        match value {
+            BatchHeaderV::V6(h) => ReferenceHeader::V6(h),
+            BatchHeaderV::V7(h) => ReferenceHeader::V7(h),
+        }
+    }
+}
+
+impl BatchHeaderV {
+    pub fn batch_hash(&self) -> B256 {
+        match self {
+            BatchHeaderV::V6(h) => h.batch_hash(),
+            BatchHeaderV::V7(h) => h.batch_hash(),
+        }
+    }
+
+    pub fn must_v6_header(&self) -> &BatchHeaderV6 {
+        match self {
+            BatchHeaderV::V6(h) => h,
+            BatchHeaderV::V7(_) => panic!("try to pick v7 header"),
+        }
+    }
+
+    pub fn must_v7_header(&self) -> &BatchHeaderV7 {
+        match self {
+            BatchHeaderV::V7(h) => h,
+            BatchHeaderV::V6(_) => panic!("try to pick v6 header"),
+        }
+    }
+}
+
+/// Defines a proving task for batch proof generation, the format
+/// is compatible with both pre-euclidv2 and euclidv2
+#[derive(Clone, serde::Deserialize, serde::Serialize)]
+pub struct BatchProvingTask {
```

Comment:
is this struct removed from zkvm-prover?

Reply:
All `xxxProvingTask` and `WrappedProof` is planned to be removed from zkvm-prover